crates/json-abi/src/item.rs

```diff
@@ -207,7 +193,58 @@ impl<'a, 'r> From<&'r AbiItem<'a>> for &'r private::AbiItem<'a> {
     }
 }
 
+impl Serialize for AbiItem<'_> {
+    #[inline]
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        <&private::AbiItem<'_>>::from(self).serialize(serializer)
+    }
+}
+
+impl<'de: 'a, 'a> Deserialize<'de> for AbiItem<'a> {
+    #[inline]
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        private::AbiItem::deserialize(deserializer).map(Into::into)
+    }
+}
```

Comment:
I forget, why do we have the `private` module?

Reply:
Comment at the top of the file

Reply:
Well, dug into this a bit more and we can get rid of it. #330